_inc/connect-button.js

```diff
@@ -72,7 +72,7 @@ jQuery( document ).ready( function( $ ) {
 		},
 		handleAuthorizationComplete: function() {
 			jetpackConnectButton.isRegistering = false;
-			location.reload();
+			window.location = jpConnect.redirectUrl;
```

Comment:
I didn't see a redirect here, still seeing the unconnected menu when seeing the plans page: 

![](https://cldup.com/4gd2oPpJgV.png)

Maybe it needs a `window.location.reload()` after all?

Reply:
There are two issues related to the suggested solution.

First is that if we initiate the `location.reload()` after reassigning the `location` value itself, the UX is poor. The page switches to the plans prompt immediately and only after a moment it gets reloaded. I guess it's because the `Router` picks up the route change early - before the browser is able to actually reload the page. And if we wanted to reload first, there wouldn't be an easy way to redirect to `#plans-prompt`.

The second issue is that - as I understand - we want to keep the user in this particular flow until they make a decision on the plan they use. So we want to remove any other distractions like masthead or sidebar menu so that the user cannot quit too early. But again, let's ask @jeffgolenski for an advice here.